modules/common/src/main/java/com/opengamma/sdk/common/ServiceInvoker.java

```diff
@@ -242,4 +246,26 @@ public void close() {
     executor.shutdown();
   }
 
+  /**
+   * Constructs the user agent to be added as a header to all HTTP requests, based on the application version.
+   *
+   * @return The user agent.
+   */
+  public static String generateUserAgent() {
+    Properties systemProperties = System.getProperties();
+    return "opengamma-sdk-java/" +
+        SDK_VERSION +
+        " (" +
+        systemProperties.getProperty("os.name") +
+        " " +
```

Comment:
The convention is to use `;` followed by a space between different elements within parentheses.